encryptionkeyrotation/README.md

```diff
@@ -0,0 +1,102 @@
+# CSI Addons Operation: EncryptionKeyRotation
+
+## Terminology
+
+| Term     | Definition                                                                            |
+| -------- | ------------------------------------------------------------------------------------- |
+| VolumeID | The identifier of the volume generated by the plugin.                                 |
+| CO       | Container Orchestration system that communicates with plugins using CSI service RPCs. |
+| SP       | Storage Provider, the vendor of a CSI plugin implementation.                          |
+| RPC      | [Remote Procedure Call](https://en.wikipedia.org/wiki/Remote_procedure_call).         |
+
+## Objective
+
+Define an extension to the CSI specification that will enable storage vendors
+(SP) to develop controllers/plugins that can rotate the Key Encryption Keys (KEK)
+of existing volumes that are encrypted.
+
+### Goals in MVP
+
+The new extension will define a procedure that
+
+- can be called for existing encrypted volumes
+- functions for Block and filesystem access mode volumes
+- interacts with the NodePlugin
+- makes it possible for the SP to rotate encrpytion keys on volumes
+
+### Non-Goals in MVP
+
+- Rotating encrpytion keys for inactive volumes
+
+## Solution Overview
+
+This specification defines an interface along with the minimum operational and
+packaging recommendations for a storage provider (SP) to implement a key rotate operation for volumes. The interface declares the RPCs that a plugin
+MUST expose.
+
+## RPC Interface
+
+- **EncryptionKeyRotate Service**: Either NodePlugin or Provisioner MAY implement this RPC.
+
+```protobuf
+syntax = "proto3";
+package encryptionkeyrotation;
+
+import "github.com/container-storage-interface/spec/lib/go/csi/csi.proto";
+import "google/protobuf/descriptor.proto";
+
+option go_package = "github.com/csi-addons/spec/lib/go/encryptionkeyrotation";
+
+// EncryptionKeyRotationController holds the RPC method for running
+// key rotation operation on the volume.
+service EncryptionKeyRotationController {
+  // EncryptionKeyRotate is a procedure that is called
+  // on the CSI Provisioner or NodePlugin
+  rpc EncryptionKeyRotate(EncryptionKeyRotateRequest)
+  returns (EncryptionKeyRotateResponse){}
+}
+```
+
+### EncryptionKeyRotateRequest
+
+```protobuf
+// EncryptionKeyRotateRequest contains the information needed to identify
+// the volume by the SP and access any backend services so that the key can be
+// rotated.
+message EncryptionKeyRotateRequest {
+  // The ID of the volume for which the key is to be rotated.
+  // This field is required
+  string volume_id = 1;
+
+  // The path where the volume is available.
+  // This field is OPTIONAL
+  // Useful if you are implementing the RPC on CSI Driver Provisioner
```

Comment:
`Useful if you are implementing the RPC on CSI Driver Provisioner` to `Useful if you are implementing the RPC on CSI Driver NodePlugin`